packages/gatsby/index.d.ts

```diff
@@ -11,24 +14,859 @@ export {
+  getNodes: Function;
+  getNode: Function;
+  getNodesByType: Function;
+  hasNodeChanged: Function;
+  reporter: Reporter;
+  getNodeAndSavePathDependency: Function;
+  cache: Cache;
+  createNodeId: Function;
+  createContentDigest: Function;
+  tracing: Tracing;
+  [key: string]: unknown;
+}
+
+export interface Actions {
+  /** @see https://www.gatsbyjs.org/docs/actions/#deletePage */
+  deletePage: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createPage */
+  createPage(args: CreatePageArgs, option?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#deletePage */
+  deleteNode(args: DeleteNodeArgs, option?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#deleteNodes */
+  deleteNodes: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createNode */
+  createNode(node: Node, options?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#touchNode */
+  touchNode: Function;
+  
+  /** @see https://www.gatsbyjs.org/docs/actions/#createNodeField */
+  createNodeField(args: CreateNodeFieldArgs, options?: ActionOptions): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createParentChildLink */
+  createParentChildLink: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createPageDependency */
+  createPageDependency: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#deleteComponentsDependencies */
+  deleteComponentsDependencies: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#replaceComponentQuery */
+  replaceComponentQuery: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#replaceStaticQuery */
+  replaceStaticQuery: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#setWebpackConfig */
+  setWebpackConfig(config: object): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#replaceWebpackConfig */
+  replaceWebpackConfig(config: object): void;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#setBabelOptions */
+  setBabelOptions: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#setBabelPlugin */
+  setBabelPlugin: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#setBabelPreset */
+  setBabelPreset: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createJob */
+  createJob: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#setJob */
+  setJob: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#endJob */
+  endJob: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#setPluginStatus */
+  setPluginStatus: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#createRedirect */
+  createRedirect: Function;
+
+  /** @see https://www.gatsbyjs.org/docs/actions/#addThirdPartySchema */
+  addThirdPartySchema: Function;
+}
+
+export interface Store {
+  dispatch: Function;
+  subscribe: Function;
+  getState: Function;
+  replaceReducer: Function;
+}
+
+export interface Reporter {
```

Comment:
The reporter got some kind of makeover so these types should suffice

```ts
type logMessageType = (format: string, ...args: any[]) => void

export interface Reporter {
  stripIndent: Function;
  format: object;
  setVerbose(isVerbose: boolean): void;
  setNoColor(isNoColor: boolean): void;
  panic(...args: any[]): void;
  panicOnBuild(...args: any[]): void;
  error(message: string, error: Error): void;
  uptime(prefix: string): void;
  success: logMessageType;
  verbose: logMessageType;
  info: logMessageType;
  warn: logMessageType;
  log: logMessageType;
  activityTimer(name: string, activityArgs: { parentSpan: object }): { start: () => void, 
status(status: string): void, end: () => void, span: object}
}
```

stripIndent uses [common-tags](https://github.com/declandewet/common-tags) & format is using [chalk](https://github.com/chalk/chalk)

Reply:
For now, it can be simpler, doing the whole `@types` dependency thing is a bit tricky ATM